content/tutorials/manage-the-clio-server/install-clio-on-ubuntu.md

```diff
@@ -0,0 +1,184 @@
+
+1. Update repositories:
+
+        sudo apt -y update
+
+2. Install utilities:
+
+        sudo apt -y install apt-transport-https ca-certificates wget gnupg
+
+3. Add Ripple's package-signing GPG key to your list of trusted keys:
+
+        sudo mkdir /usr/local/share/keyrings/
+        wget -q -O - "https://repos.ripple.com/repos/api/gpg/key/public" | gpg --dearmor > ripple-key.gpg
+        sudo mv ripple-key.gpg /usr/local/share/keyrings
+
+
+4. Check the fingerprint of the newly-added key:
+
+        gpg /usr/local/share/keyrings/ripple-key.gpg
+
+    The output should include an entry for Ripple such as the following:
+
+        gpg: WARNING: no command supplied.  Trying to guess what you mean ...
+        pub   rsa3072 2019-02-14 [SC] [expires: 2026-02-17]
+            C0010EC205B35A3310DC90DE395F97FFCCAFD9A2
+        uid           TechOps Team at Ripple <techops+rippled@ripple.com>
+        sub   rsa3072 2019-02-14 [E] [expires: 2026-02-17]
+
+
+    In particular, make sure that the fingerprint matches. (In the above example, the fingerprint is on the third line, starting with `C001`.)
+
+4. Add the appropriate Ripple repository for your operating system version:
+
+        echo "deb [signed-by=/usr/local/share/keyrings/ripple-key.gpg] https://repos.ripple.com/repos/rippled-deb focal stable" | \
+            sudo tee -a /etc/apt/sources.list.d/ripple.list
+
+    The above example is appropriate for **Ubuntu 20.04 Focal Fossa**. 
+
+
+5. Fetch the Ripple repository.
+
+        sudo apt -y update
+
+6. Install the Clio software package:
+
+        sudo apt -y install rippled
+
+7. Run `./clio_server config.json`.
+
+8. A Clio server needs to access a `rippled` server to run succesfully. To enable communication between the servers, the config files of Clio and `rippled` need to share the following information.
+
+    1. Update the Clio server's config file with the following information:
+        
+        * The IP of `rippled` server.
+        * The port on which `rippled` is accepting unencrypted WebSocket connections.
+        * The port on which `rippled` is handling gRPC requests.
+
+                    "etl_sources":
+                    [
+                        {
+                            "ip":"127.0.0.1",
+                            "ws_port":"6006",
+                            "grpc_port":"50051"
+                        }
+                    ]
+
+        **Note** You can use multiple `rippled` servers as a data source by add more entries to the `etl_sources` section. Clio will load balance requests across the servers specified in the list. As long as one `rippled` server is up and synced, Clio will continue to extract validated ledgers.
+
+    2. Update the `rippled` server's config file with the following information:
+        
+        * Open a port to accept unencrypted websocket connections. 
+        * Open a port to handle gRPC requests and specify the IP(s) of Clio server(s) in the `secure_gateway` entry.
```

Comment:
Is this an example of setting `port_grpc` and `ws_public` in rippled? if so, this is how you do that in `rippled.cfg` 
```cfg
[port_grpc]
port = 50051
ip = 0.0.0.0
secure_gateway = <clio_ip_address>
```

```cfg
[port_ws_public]
port = 6005
ip = 0.0.0.0
protocol = ws
```

Reply:
Added to the file